src/index.js

```diff
@@ -5,9 +5,37 @@
 const fs = require('fs');
 const feedQueue = require('./feed-queue');
 const feedWorker = require('./feed-worker');
+const parentLogger = require('../src/lib/logger');
+
+const log = parentLogger.child({ module: 'main' });
 
 // Start the web server
-require('./backend/web/server');
+const server = require('./backend/web/server');
+
+/**
+ * Stops the Redis Queue, closes all connections gracefuly
+ */
+function shutDown() {
+  log.info('Received kill signal, shutting down gracefully');
+
+  // Close the Redis Queue
+  feedQueue.close();
```

Comment:
I think we should probably `await` this, or see if you can pass a callback.  At the very least, we should log errors that this produces.

Reply:
@humphd Will using `await feedQueue.close()` be enough?

Reply:
That's so tricky. If we want to do it `await`, maybe we want also `setTimeout` before `await feedQueue.close();`

Reply:
Here's my suggestion:

```js
setTimeout() => {
    log.error('Could not close connections in time, forcefully shutting down');
    process.exit(1);
}, 10000);

// Try to shut down 
Promise.all([
  // Shutdown the queue
  feedQueue.close().catch(err => log.error('Unable to close feed queue gracefully', err),
  // Shutdown the web server - we could also use util.promisify if you want
  new Promise((resolve, reject) => server.close(err => {
    if(err) {
      log.error('Unable to close web server gracefully', err);
      reject(err);
    } else {
      resolve();
    }
   })
])
.then(() => process.exit(0))
.catch(err => log.error(error));
```

Reply:
I noticed this while looking up something for your code: https://expressjs.com/en/advanced/healthcheck-graceful-shutdown.html.

We could do one of those in a later PR too.  I don't want to expand this PR forever, but it's neat to see.